### PR TITLE
feat: added down arrow to show the end of transactions

### LINKED ISSUE #812: feat: scroll to bottom arrow in history fragment
**Summary:**

We need to have a scroll to bottom down arrow in transaction history fragment as it will help the users to check their bottom most chats

### PR DESCRIPTION
**Pull request by Bhavnaharitsa**
Fixes: #812 
Added down arrow in HistoryFragment to show the end of histories like in whatsapp or such apps which leads to the bottom of the page or chats
- [.] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [.] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [.] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by Bhavnaharitsa** [comment]
@luckyman20 Please review

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/HistoryContract.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
         void showTransactionDetailDialog(int transactionIndex, String accountNumber);
 
         void showHistoryFetchingProgress();
+
+        void scrollDownBottom();
```
Comment (rahul-jha98) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/HistoryContract.java]: The name of the function should reflect what you are doing in the function. In this case scrollDownBottom gives the impression that the function will be called once you need to scroll to bottom, but that's not the case in the actual implementation. 
Instead a more explanatory function name, like say setupScrollDownBottomButton will lead to better readability of code.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java [hunk 2]
```diff
@@ -87,6 +92,29 @@ private void setupUi() {
...
+    public void scrollDownBottom() {
+        rvHistory.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
+                super.onScrollStateChanged(recyclerView, newState);
+                if (!recyclerView.canScrollVertically(1)) {
+                    Toast.makeText(getContext(), R.string.end_of_transactions,
+                            Toast.LENGTH_SHORT).show();
+                } else {
+                    scrollDownButton.setVisibility(View.VISIBLE);
```
Comment (rahul-jha98) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java]: Since you have written this line I guess you intended to hide the button when the user is at the bottom of the RecyclerView. 
If that's the case make sure you hide the button in the if block.

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java [hunk 3]
```diff
@@ -87,6 +92,29 @@ private void setupUi() {
...
+                }
+            }
+        });
+        scrollDownButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                int length = mHistoryAdapter.getItemCount();
+                rvHistory.scrollToPosition(length - 1);
```
Comment (rahul-jha98) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java]: Just a suggestion here :) 
Although I have not seen this in action in the actual app, but generally scrollToPosition leads to abrupt change in the UI. 
I would suggest trying smoothScrollToPosition which leads to decent animation while scrolling to the bottom.

**Review by rahul-jha98 — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java (hunk 2 above)
Comment (Bhavnaharitsa) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java]: You're right, I have not hidden it. Thank you for the heads up!!
Shall do it and squash

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java (hunk 3 above)
Comment (Bhavnaharitsa) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java]: Okay!

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/HistoryContract.java (hunk 1 above)
Comment (Bhavnaharitsa) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/HistoryContract.java]: What say of the name scrollDownBottomButton()?

**Comment by refactor-droidyy** [comment]
@Bhavnaharitsa Please attach a proper gif reflecting the UI change made.

**Comment by Bhavnaharitsa** [comment]
@refactor-droidyy Yes, sure! I was about to but then when I was about to record, the server went down and it came, couldn't fetch transactions. 
Let me try once and attach asap :)

**Comment by Bhavnaharitsa** [comment]
Screenshot_20200317-172307_Mifos Pay

@refactor-droidyy Here is a screen shot, shall attach video soon

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/HistoryContract.java (hunk 1 above)
Comment (Bhavnaharitsa) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/HistoryContract.java:35]: @rahul-jha98 updated the changes, if you want to have a look :)

Comment (rahul-jha98) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/HistoryContract.java:35]: The other changes looks fine to me. But are you sure you have renamed the function. I see its still named the same as before. Please look into this.

Comment (Bhavnaharitsa) [reply mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/HistoryContract.java]: @rahul-jha98 Sorry sir, seems like I have missed it. I have changed it now :)

**Comment by Bhavnaharitsa** [comment]
@luckyman20 @shiv07tiwari @naman14 Please review it

**Comment by devansh-299** [comment]
> Screenshot_20200317-172307_Mifos Pay
> 
> @refactor-droidyy Here is a screen shot, shall attach video soon

@Bhavnaharitsa can you please attach a gif for showing the changes you did

**Comment by Bhavnaharitsa** [comment]
> @refactor-droidyy Here is a screen shot, shall attach video soon
> 
> @Bhavnaharitsa can you please attach a gif for showing the changes you did

yes, once history starts showing

**Comment by devansh-299** [comment]
> > @refactor-droidyy Here is a screen shot, shall attach video soon
> > @Bhavnaharitsa can you please attach a gif for showing the changes you did
> 
> yes, once history starts showing

@Bhavnaharitsa transaction history is working fine for me, is your `API_ENDPOINT` in `mobile-wallet/core/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/BaseURL.java` set to demo.mifos.io?

**Comment by Bhavnaharitsa** [comment]
ezgif com-video-to-gif
@devansh-299 you were right, the end point was different. I changed it now :)
Please review @shiv07tiwari @luckyman20 @devansh-299 @rachittt

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java [hunk 4]
```diff
@@ -87,6 +92,30 @@ private void setupUi() {
...
     }
 
+    @Override
+    public void setUpScrollDownBottomButton() {
+        rvHistory.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
+                super.onScrollStateChanged(recyclerView, newState);
+                if (!recyclerView.canScrollVertically(1)) {
+                    Toast.makeText(getContext(), R.string.end_of_transactions,
```
Comment (s-ayush2903) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java:102]: Why not using Toaster?

File: mifospay/src/main/res/layout/fragment_history.xml [hunk 5]
```diff
@@ -18,6 +18,18 @@
         android:layout_height="match_parent"
         android:scrollbars="vertical" />
 
+    <ImageView
```
Comment (s-ayush2903) [inline mifospay/src/main/res/layout/fragment_history.xml:21]: When using CoordinatorLayout then, why ImageView, not fab?

File: mifospay/src/main/res/layout/fragment_history.xml [hunk 6]
```diff
@@ -18,6 +18,18 @@
...
+        android:layout_width="@dimen/value_30dp"
+        android:layout_height="@dimen/value_30dp"
+        android:layout_gravity="bottom|end"
+        android:layout_marginEnd="20dp"
```
Comment (s-ayush2903) [inline mifospay/src/main/res/layout/fragment_history.xml:25]: Please refrain hardcoding

File: mifospay/src/main/res/layout/fragment_history.xml [hunk 7]
```diff
@@ -18,6 +18,18 @@
...
+        android:layout_marginRight="20dp"
```
Comment (s-ayush2903) [inline mifospay/src/main/res/layout/fragment_history.xml:26]: Same

File: mifospay/src/main/res/layout/fragment_history.xml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.design.widget.CoordinatorLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<android.support.design.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (s-ayush2903) [inline mifospay/src/main/res/layout/fragment_history.xml:2]: Is this change really needed?

File: mifospay/src/main/res/layout/fragment_history.xml [hunk 9]
```diff
@@ -1,7 +1,7 @@
...
     android:layout_width="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (s-ayush2903) [inline mifospay/src/main/res/layout/fragment_history.xml:4]: Same

**Comment by s-ayush2903** [comment]
Also, please do resize the screen snips or visuals while attaching

**Comment by devansh-299** [comment]
@Bhavnaharitsa  just a suggestion, on reaching the bottom of the list, this same functionality/arrow can be overloaded to take users back to the top with a top arrow.

File: mifospay/src/main/res/layout/fragment_history.xml (hunk 8 above)
Comment (Bhavnaharitsa) [reply mifospay/src/main/res/layout/fragment_history.xml:2]: It is not a change I made manually. Its GitHub glitch

Comment (devansh-299) [reply mifospay/src/main/res/layout/fragment_history.xml:2]: > It is not a change I made manually. Its GitHub glitch

Actually these are because of Android Studio's auto linting. You can fix this by editing this file in other text editor or manually changing your Android studio specifications

Comment (s-ayush2903) [reply mifospay/src/main/res/layout/fragment_history.xml:2]: @Bhavnaharitsa Ma'am, github isn't glitchy, as @devansh-299 pointed out, you can change this in editor preferences and by setting codestyle to the `Project` Style indentations

Comment (Bhavnaharitsa) [reply mifospay/src/main/res/layout/fragment_history.xml:2]: Okay, shall keep a note of it

**Comment by Bhavnaharitsa** [comment]
> @Bhavnaharitsa just a suggestion, on reaching the bottom of the list, this same functionality/arrow can be overloaded to take users back to the top with a top arrow.

Let me see

**Comment by Bhavnaharitsa** [comment]
@luckyman20  @rachittt I apologize that this branch is lost in my local directory due to some viruses. However, I have solved the virus thing and now shall make a new pr. I know that this practise is not good in open source but I will make sure I don't repeat it
